PowerShell.Common.props

```diff
@@ -96,8 +96,8 @@
     <Copyright>(c) Microsoft Corporation.</Copyright>
 
     <TargetFramework>net5.0</TargetFramework>
-    <LangVersion>8.0</LangVersion>
-    <PublishReadyToRun>true</PublishReadyToRun>
+    <LangVersion>preview</LangVersion>
```

Comment:
I have a concern that a language feature in preview may not make to the official release, so what if we start to use a syntax in code and that feature is pulled out?

Reply:
One feature I want is new `is not`. It is unbelievable that it will be removed.
Other new C# features are very specific and are not of interest at the moment. We will reject its till release in code reviews.
We already did this when Span<T> appeared.

Reply:
I think it's a valid concern, but the risk is low, especially if we're judicious about which features we use, as @iSazonov says. We're already taking .NET preview versions, and I'd say the risk of needing to roll those back is much greater.

Reply:
As long as we can for a non-preview version of PowerShell ensure that we pin to a non-preview version of the C# language then I'd be happy with this change.

But it needs extra rigour to ensure that we don ship a 7.0.3 / 7.1.0 that uses preview language versions.

I expect most of that rigour is already in place, but still needed to be called out, just in case

Reply:
Well we don't synchronise to C# language version releases like .NET releases, so we are forced to retain preview language versions for stable releases.

But that's not uncommon, and unlike .NET preview releases, the language features all compile down to the same things. Using a preview C# version will affect the development, but is unlikely to affect the product.

I think the real risk is that a preview feature is taken out of the language because it doesn't meet some bar, at which point we'll need to go and change all the code that uses that feature in order to build again.

I'll add that it makes backporting harder theoretically, but so would just incrementing the language version to another stable version.

Reply:
Is that due to the difference in release cycle of c# vs the release cycle of .net as a whole? 

I would have thought that from a supportability & security perspective using a preview version of the c# language not a stable version would have the potential to cause upsets & augments about support further down the line. Is this not the case?

I'm not sure I agree on this, I think it's more likely to be uncommon but this isn't really my area of expertise 
> But I don't think it's uncommon to build a stable product on a preview language version

would be more like this
> Typically you'd build a preview product using preview language versions and then release the stable version of the product when the stable version of the language is released.

I guess the concern I have is likely one that you've already covered but should we not be pushing to make use of non-preview functionality in the end resulting product & instead wait a little for the Language to move from Preview version to the fully Supported version. 
Or is that where the x.0.1 release would come in play to move from the Language preview to the actual new Language version?

Not looking to place a blocker, more wanting to understand the moving components better that's all.

Reply:
From what I can gather, this is actually a moot discussion because C# 9 should stabilise with .NET 5. The only risk that remains then is that something we depend on in preview does not progress to stable. However, the risk here is identical to relying on a preview .NET API -- we would need to revert to using the old implementation if it happened.

Reply:
Thanks for the clarification @rjmholt that makes a lot of sense :-)

Reply:
I've followed up with the .NET team on this to understand if our assumptions here can be relied on. I think until that's the case, it's fair for us to hold this PR

Reply:
Ok I’ve confirmed that C# 9 is planned to stabilise with .NET 5 — so our only remaining concern is the tolerance of removed preview features

Reply:
Current plan is only modernizing our code with `is not null` feature. No plans exist to use other new features in near month or two before C# 9.0 will be stabilized.